dandiapi/api/tasks.py

```diff
@@ -56,7 +63,81 @@ def write_yamls(version_id: int) -> None:
         logger.info('%s already exists, deleting it', assets_yaml_path)
         storage.delete(assets_yaml_path)
     logger.info('Saving %s', assets_yaml_path)
-    assets_yaml = YAMLRenderer().render([asset.metadata.metadata for asset in version.assets.all()])
+    assets_yaml = YAMLRenderer().render(
+        [asset.generate_metadata(version) for asset in version.assets.all()]
+    )
     storage.save(assets_yaml_path, ContentFile(assets_yaml))
 
     logger.info('Wrote dandiset.yaml and assets.yaml for version %s', version_id)
+
+
+def format_as_index(indices):
+    """Render a JSON schema path as a series of indices."""
+    if not indices:
+        return ''
+    return '[%s]' % ']['.join(repr(index) for index in indices)
```

Comment:
This might make it a bit more readable, just a suggestion though feel free to ignore
```suggestion
    return '[' + ']['.join(repr(index) for index in indices) + ']'
```